src/main/java/ca/on/oicr/pineryreports/reports/impl/LanesBillingReport.java

```diff
@@ -0,0 +1,464 @@
+package ca.on.oicr.pineryreports.reports.impl;
+
+import static ca.on.oicr.pineryreports.util.GeneralUtils.byEndedBetween;
+import static ca.on.oicr.pineryreports.util.GeneralUtils.timeStringToYyyyMmDd;
+import static ca.on.oicr.pineryreports.util.SampleUtils.isRnaLibrary;
+import static ca.on.oicr.pineryreports.util.SampleUtils.mapSamplesById;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.ParseException;
+
+import com.google.common.collect.Sets;
+
+import ca.on.oicr.pinery.client.HttpResponseException;
+import ca.on.oicr.pinery.client.PineryClient;
+import ca.on.oicr.pineryreports.data.ColumnDefinition;
+import ca.on.oicr.pineryreports.reports.TableReport;
+import ca.on.oicr.pineryreports.util.CommonOptions;
+import ca.on.oicr.ws.dto.InstrumentDto;
+import ca.on.oicr.ws.dto.InstrumentModelDto;
+import ca.on.oicr.ws.dto.RunDto;
+import ca.on.oicr.ws.dto.RunDtoPosition;
+import ca.on.oicr.ws.dto.RunDtoSample;
+import ca.on.oicr.ws.dto.SampleDto;
+
+public class LanesBillingReport extends TableReport {
+  
+  private static class DetailedObject {
+    private final String instrumentModel;
+    private final String instrumentName;
+    private final String laneContents;
+    private final String laneNumber;
+    private final String project;
+    private final BigDecimal projLibsPercent; // percentage of libs in lane which belong to given project
+    private final String runEndDate;
+    private final String runName;
+    private final String runStatus;
+    
+    public DetailedObject (String runEndDate, String instrumentName, String instrumentModel, String runName, 
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)